sale_import_base/models/sale_channel_importer.py

```diff
@@ -163,17 +178,34 @@ def _prepare_sale_line_vals(self, data, sale_order):
         return [self._prepare_sale_line(line, sale_order) for line in data["lines"]]
 
     def _prepare_sale_line(self, line_data, sale_order):
+        channel = self.chunk_id.reference
+        company_id = channel.company_id
+
         product = self.env["product.product"].search(
-            [("default_code", "=", line_data["product_code"])]
+            [
+                ("default_code", "=", line_data["product_code"]),
+                ("product_tmpl_id.company_id", "=", company_id.id),
```

Comment:
I think it is not ok, because we could have products shared between companies, in that case, the company_id is not set and we should find the product.

I believe we should at least check if company is the same as the channel OR is empty